shared/editor/nodes/ToggleBlock.ts

```diff
@@ -0,0 +1,28 @@
+import { NodeSpec, NodeType } from "prosemirror-model";
+import toggleWrap from "../commands/toggleWrap";
+import Node from "./Node";
+
+export default class ToggleBlock extends Node {
+  folded = false;
+
+  get name() {
+    return "toggle_block";
+  }
+
+  get schema(): NodeSpec {
+    return {
+      content: "toggle_head toggle_body?",
```

Comment:
`toggle_block` is composed of `toggle_head` & `toggle_body` now. This choice was made so that it's convenient to apply decoration to hide _only_ `toggle_body`. Also, the `toggle_body` can be easily _indented_ to distinguish it from whatever is outside of toggle block.
<img width="1268" alt="Screenshot 2025-02-19 at 9 04 32 PM" src="https://github.com/user-attachments/assets/c52ce355-455b-449a-bce6-088f0ea37539" />

Reply:
Okay, will have to test! Not totally against this solution – though I feel that toggle_body isn't _strictly_ necessary as we can always hide all the children except the first (the heading)

Reply:
Also, simplifies the problem indenting toggle block content. Not strictly necessary though, agreed!